lib/SILGen/SILGenExpr.cpp

```diff
@@ -4181,6 +4181,15 @@ SILGenModule::emitKeyPathComponentForDecl(SILLocation loc,
       return false;
     }
 
+    // Don't need to use the external component for a standard library type that
+    // is not resilient in this context. Standard library types may have public
+    // accessors, but if they are not resilient, they don't need an external
+    // key path component and don't have property descriptors.
+    if (baseDecl->getModuleContext()->isStdlibModule() &&
```

Comment:
Why is this specific to the standard library?

Reply:
I was trying to address types like Int - is there another way?

Reply:
(I just understood your question, so trying again:) For static keypaths, we're emitting property descriptors for keypaths referencing properties declared in other modules. For types like `Int`, these property descriptors seem to not exist in IRGen. My initial understanding was they should not be emitted if they're in a Swift module, and so tried to check for `isStdLibModule`, but I may have misunderstood what `SwiftModule` means.

Reply:
> For types like Int, these property descriptors seem to not exist in IRGen. 

Why is that, though? :)

Reply:
>> For types like Int, these property descriptors seem to not exist in IRGen.
> 
> Why is that, though? :)

They should exist and did not because the code was preventing their emission for types conforming to protocol extensions. 

I originally assumed that "native swift" modules like the standard library shouldn't be considered a different module than `SwiftModule` and then wrote a fix that prevented descriptor emission for keypaths with standard library types, but that was not a correct understanding of descriptors. Even though I didn't understand your question immediately, it did make me rethink why I was handling the standard library module explicitly in this way and look for clarification on descriptor emission here.